config/systempatches/patches.inc

```diff
@@ -200,36 +195,38 @@ function bootup_apply_patches() {
 function patch_add_shellcmd() {
 	global $config;
 	$a_earlyshellcmd = &$config['system']['earlyshellcmd'];
-	if (!is_array($a_earlyshellcmd)) {
-		$a_earlyshellcmd = array();
-	}
 	$found = false;
-	foreach ($a_earlyshellcmd as $idx => $cmd) {
-		if (stristr($cmd, "apply_patches.php")) {
-			$found = true;
+	if (is_array($a_earlyshellcmd)) {
+		foreach ($a_earlyshellcmd as $idx => $cmd) {
+			if (stristr($cmd, "apply_patches.php")) {
+				$found = true;
+			}
 		}
+	} else {
+		$a_earlyshellcmd = array();
```

Comment:
It does not need to be explicitly created as an array in that case. It will be turned into an array during the assignment with []. The only reason it had to be explicitly declared was to make foreach happy.